### PR TITLE
Cargo publish CI

### LINKED ISSUE #236: Improve release process
Hi there,

Where is the source of `refinery` version `0.8.5`? The `master` branch's `refinery/Cargo.toml` is at `0.8.4`, however crates.io is at `0.8.5`.

Thanks!

### PR DESCRIPTION
**Pull request by TobiasDeBruijn**
Closes #236 

This PR adds a CI that's run on a Git tag, it'll publish the crate to crates.io.
The `CARGO_TOKEN` secret must be set to a crates.io access token. This Action uses GitHub secrets for the token.

The action does not validate that the Git tag matches the configured version in the crate, nor does it run tests. The tag is assumed to be validated manually. Tests should be run on CI before a tag is created.

Thanks.

### CODE REVIEW

File: .github/workflows/cargo-publish.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - uses: actions-rs/cargo@v1
+      env:
+        CARGO_TOKEN: ${{ secrets.CARGO_TOKEN }}
+      with:
+        command: publish
+        args: --token $CARGO_TOKEN
```
Comment (jxs) [inline .github/workflows/cargo-publish.yaml:30]: I think we need to publish each crate individually though, either by cd'ing into each crate and publishing it, or with `-p` on the root

**Review by jxs — COMMENTED** [review]
thanks! :heart: 
Just one remark

File: .github/workflows/cargo-publish.yaml (hunk 1 above)
Comment (TobiasDeBruijn) [reply .github/workflows/cargo-publish.yaml:30]: What's the manual release process currently like? I'll need to mirror that.

Comment (jxs) [reply .github/workflows/cargo-publish.yaml:30]: right, sorry should have wrote that on the first comment. Basically what I do is in order release `refinery-core` first, then `refinery-macros` then `refinery` and lastly `refinery-cli` with:

```bash
$ cargo publish --all-features refinery-core
$ cargo publish --all-features refinery-macros
$ cargo publish --all-features refinery
$ cargo publish --all-features refinery_cli
```

Comment (TobiasDeBruijn) [reply .github/workflows/cargo-publish.yaml:30]: Check. That should be easy enough. I'll update the PR tomorrow.

**Review by jxs — APPROVED** [review]

**Comment by jxs** [comment]
thanks again Tobias! :bow:

**Pull request merged**